python/ray/tune/result_grid.py

```diff
@@ -240,3 +240,10 @@ def _trial_to_result(self, trial: Trial) -> Result:
             best_checkpoints=best_checkpoints,
         )
         return result
+
+    def __repr__(self) -> str:
+        all_results_repr = ""
+        for result in self:
+            result_repr = "  " + result.__repr__().replace("\n", "\n  ")
+            all_results_repr += f"{result_repr},\n"
```

Comment:
1. Can we use a list comprehension here?
2. The replace looks a bit fishy. Can we maybe instead use a private method in `Result` for both `Result.__repr__` and `ResultGrid.__repr__`?